src/main/java/org/kohsuke/github/GitHub.java

```diff
@@ -558,7 +558,7 @@ public GHRepository getRepository(String name) throws IOException {
      * @throws IOException
      *             the io exception
      */
-    public GHRepository getRepositoryById(String id) throws IOException {
+    public GHRepository getRepositoryById(long id) throws IOException {
```

Comment:
Please make this a separate PR and keep the existing method but make it deprecated. 
Also, add tests for both the old and new methods.  Thanks!